anesthetic/plot.py

```diff
@@ -505,59 +496,52 @@ def kde_plot_1d(ax, data, *args, **kwargs):
         matplotlib matplotlib.axes.Axes.plot command)
 
     """
+    kwargs = normalize_kwargs(
+        kwargs,
+        dict(linewidth=['lw'], linestyle=['ls'], color=['c'],
+             facecolor=['fc'], edgecolor=['ec']))
+
     if len(data) == 0:
         return np.zeros(0), np.zeros(0)
 
     if data.max()-data.min() <= 0:
         return
 
-    kwargs = normalize_kwargs(
-        kwargs,
-        dict(linewidth=['lw'], linestyle=['ls'], color=['c'],
-             facecolor=['fc'], edgecolor=['ec']))
+    weights = kwargs.pop('weights', None)
+    if weights is not None:
+        data = data[weights != 0]
+        weights = weights[weights != 0]
 
+    ncompress = kwargs.pop('ncompress', 1000)
     bw_method = kwargs.pop('bw_method', None)
     levels = kwargs.pop('levels', [0.95, 0.68])
-
-    xmin = kwargs.pop('xmin', None)
-    xmax = kwargs.pop('xmax', None)
-    weights = kwargs.pop('weights', None)
-    ncompress = kwargs.pop('ncompress', 1000)
     density = kwargs.pop('density', False)
+
     cmap = kwargs.pop('cmap', None)
     color = kwargs.pop('color', (next(ax._get_lines.prop_cycler)['color']
                                  if cmap is None else cmap(0.68)))
     facecolor = kwargs.pop('facecolor', False)
-
     if 'edgecolor' in kwargs:
         edgecolor = kwargs.pop('edgecolor')
         if edgecolor:
             color = edgecolor
     else:
         edgecolor = color
 
-    q = kwargs.pop('q', '5sigma')
+    q = kwargs.pop('q', 5)
     q = quantile_plot_interval(q=q)
 
-    if weights is not None:
-        data = data[weights != 0]
-        weights = weights[weights != 0]
-
     x, w = sample_compression_1d(data, weights, ncompress)
     kde = gaussian_kde(x, weights=w, bw_method=bw_method)
     p = kde(x)
     p /= p.max()
     i = ((x > quantile(x, q[0], w)) & (x < quantile(x, q[1], w)))
```

Comment:
Could we raise an issue to improve the quantile treatment for defining x's, so that we drive 'five sigma quantiles' by iso-probability credibility regions, rather than cdfs?

Reply:
See issue #213